### PR TITLE
Meet review comments for #1265

### PR DESCRIPTION
**Pull request by hkvision**
#1265 Address new comments from @dding3

### CODE REVIEW

File: zoo/src/main/scala/com/intel/analytics/zoo/examples/vnni/bigdl/ImageNetInference.scala [hunk 1]
```diff
@@ -47,16 +47,20 @@ object ImageNetInference {
...
+        .text("The number of partitions to cut the dataset into")
+        .action((x, c) => c.copy(partitionNum = x))
     }
     parser.parse(args, ImageNetInferenceParams()).map(param => {
       val sc = NNContext.initNNContext("ImageNet2012 with Int8 Inference Example")
       val images = ImageSet.readSequenceFiles(param.folder, sc)
+      // If the original partitionNum of sequence files is too large, then the
+      // total batchSize we calculate (partitionNum * batchPerPartition) would be
+      // too large for inference.
+      images.rdd = images.rdd.coalesce(param.partitionNum)
```
Comment (dding3) [inline zoo/src/main/scala/com/intel/analytics/zoo/examples/vnni/bigdl/ImageNetInference.scala]: Can we replace `coalesce` as `ImageSet.readSequenceFiles(param.folder, sc, param.partitionNum)`

File: zoo/src/main/scala/com/intel/analytics/zoo/feature/image/ImageSetToSample.scala [hunk 2]
```diff
@@ -58,7 +61,10 @@ class ImageSetToSample[T: ClassTag](inputKeys: Array[String] = Array(ImageFeatur
             require(target.isInstanceOf[Tensor[T]], s"the target $key should be tensor")
             Some(target.asInstanceOf[Tensor[T]])
           }
-          else None
+          else {
+            logger.debug(s"The ImageFeature doesn't contain targetKey $key, ignoring it")
```
Comment (dding3) [inline zoo/src/main/scala/com/intel/analytics/zoo/feature/image/ImageSetToSample.scala]: logger.warning ?

File: zoo/src/main/scala/com/intel/analytics/zoo/feature/image/ImageSetToSample.scala [hunk 3]
```diff
@@ -68,7 +74,7 @@ class ImageSetToSample[T: ClassTag](inputKeys: Array[String] = Array(ImageFeatur
       case e: Exception =>
         e.printStackTrace()
         val uri = feature.uri()
-        ImageSetToSample.logger.warn(s"The conversion from ImageFeature to Sample fails for $uri")
+        logger.warn(s"The conversion from ImageFeature to Sample fails for $uri")
```
Comment (dding3) [inline zoo/src/main/scala/com/intel/analytics/zoo/feature/image/ImageSetToSample.scala]: logger.error ?

File: zoo/src/main/scala/com/intel/analytics/zoo/examples/vnni/bigdl/ImageNetInference.scala [hunk 4]
```diff
@@ -47,16 +47,20 @@ object ImageNetInference {
         .text("The path to the pre-trained int8 ResNet50 model snapshot")
         .action((x, c) => c.copy(model = x))
         .required()
-      opt[Int]('b', "batchSize")
-        .text("The total batch size for inference")
-        .action((x, c) => c.copy(batchSize = x))
+      opt[Int]("partitionNum")
+        .text("The number of partitions to cut the dataset into")
```
Comment (dding3) [inline zoo/src/main/scala/com/intel/analytics/zoo/examples/vnni/bigdl/ImageNetInference.scala]: dataset partition number

File: zoo/src/main/scala/com/intel/analytics/zoo/feature/image/ImageSetToSample.scala (hunk 2 above)
Comment (hkvision) [reply zoo/src/main/scala/com/intel/analytics/zoo/feature/image/ImageSetToSample.scala]: Changed.

File: zoo/src/main/scala/com/intel/analytics/zoo/feature/image/ImageSetToSample.scala (hunk 3 above)
Comment (hkvision) [reply zoo/src/main/scala/com/intel/analytics/zoo/feature/image/ImageSetToSample.scala]: Sure.

**Comment by hkvision** [comment]
jenkins passed: http://10.239.47.211:18888/job/ZOO-PR-Validation/2236/

File: zoo/src/main/scala/com/intel/analytics/zoo/examples/vnni/bigdl/ImageNetInference.scala (hunk 1 above)
Comment (hkvision) [reply zoo/src/main/scala/com/intel/analytics/zoo/examples/vnni/bigdl/ImageNetInference.scala]: readSequenceFiles takes minPartitions and thus the result RDD doesn't necessarily has partitionNum. I added a check and coalesce if the actual partition is not as user specified.

**Pull request merged**